### PR TITLE
Feat: Add flag to disable debug file upload

### LINKED ISSUE #4056: Request to have a specific flag to disable upload debug symbol build phase
### Specs
SDK Version: @sentry/react-native: 5.28.0
RN Version: 0.74.5
expo: 51.0.26

OS:
- [ ] Windows
- [x] MacOs
- [ ] Linux

Platform:
- [x] iOS
- [ ] Android

Are you using Expo?
- [x] Yes
- [ ] No

Are you using sentry.io or on-premise?
- [x] sentry.io (SaaS)
- [ ] on-premise

---

### Context
Our expo managed react native application consists of hardening process that is done after an `.ipa` is generated. We use the expo plugin to connect the application with Sentry. This also adds `Upload Debug Symbol for Sentry` step into the build phases. 

The hardening process will make some changes to the Debug Information Files, so we are required to use upload this new one into Sentry Dashboard instead of the one uploaded during the build phase.

---

### Current workaround
We are required to patch the package and add a Release check under `sentry-dexcode-debug-files.sh` to disable the build phase step.

---

### Request
To have a specific flag to disable this step entirely, something like `SENTRY_DISABLE_DIF_UPLOAD`. Instead of using the general `SENTRY_DISABLE_AUTO_UPLOAD`. 

Was the reasoning to coupled both sourcemap and DIF upload to a single flag, under the assumption that if a dev makes some changes towards it they would be changing both js and native bundles?

### PR DESCRIPTION
**Pull request by lucas-zimerman**
## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

A new flag that can be used to disable the upload of native debug files.

## :bulb: Motivation and Context

Users want to disable native debug file upload without altering the flag SENTRY_DISABLE_AUTO_UPLOAD, this introduces a new flag called SENTRY_DISABLE_XCODE_DEBUG_UPLOAD for disabling the native debug files upload on Xcode.

## :green_heart: How did you test it?

running the script locally with the flag

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [ ] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

Close #4056

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
The flag looks good, I'm just curious, what is the use case? When a dev would like to upload source maps but no native files?

Would it make sense to have it also for Android, as the generic `SENTRY_DISABLE_AUTO_UPLOAD` works for both iOS and Android.

**Comment by lucas-zimerman** [comment]
> The flag looks good, I'm just curious, what is the use case? When a dev would like to upload source maps but no native files?

I can see some specific use cases only:
- Focusing only on JavaScript source maps.
- Saving storage on self-hosted servers (and again focusing on JS source maps)
- Speed up build process (despite being fast) for internal production ready builds.

The case of https://github.com/getsentry/sentry-react-native/issues/4056 was very specific, and I don't other use cases other than the linked issue have.

**Review by antonis — APPROVED** [review]
The implementation LGTM 🚀

File: packages/core/sentry.gradle [hunk 1]
```diff
@@ -4,7 +4,8 @@ import java.util.regex.Matcher
 import java.util.regex.Pattern
 
 project.ext.shouldSentryAutoUpload = { ->
-  return System.getenv('SENTRY_DISABLE_AUTO_UPLOAD') != 'true'
+  return System.getenv('SENTRY_DISABLE_AUTO_UPLOAD') != 'true' ||
+         System.getenv('SENTRY_DISABLE_NATIVE_DEBUG_UPLOAD') != 'true'
```
Comment (antonis) [inline packages/core/sentry.gradle:8]: Hey @lucas-zimerman 👋 
While checking some [a CI checks issue on a dependandabot PR](https://github.com/getsentry/sentry-react-native/pull/4258#issuecomment-2468504313) I revisited the Android logic here. I think for the new flag to work as on iOS we should change `||` with `&&`. Wdyt?

```
project.ext.shouldSentryAutoUpload = { ->
  return System.getenv('SENTRY_DISABLE_AUTO_UPLOAD') != 'true' &&
         System.getenv('SENTRY_DISABLE_NATIVE_DEBUG_UPLOAD') != 'true'
}
```

or 
```
project.ext.shouldSentryAutoUpload = { ->
  return !(System.getenv('SENTRY_DISABLE_AUTO_UPLOAD') == 'true' ||
         System.getenv('SENTRY_DISABLE_NATIVE_DEBUG_UPLOAD') == 'true')
}
```

**Pull request merged**